src/axolotl/utils/fim_rank.py

```
@@ -0,0 +1,306 @@
+    return result
+
+
+def _allocate_ranks(
+    importance: dict[str, float],
+    base_r: int,
+    r_min: int,
+    r_max: int,
+) -> dict[str, int]:
+    """Allocate integer ranks proportional to importance under a fixed budget.
+
+    Budget = n_layers × base_r (mean rank preserved).
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)